octobot_tentacles_manager/base_worker/tentacle_worker.py

```diff
@@ -0,0 +1,199 @@
+
+        self.fetched_for_requirements_tentacles = []
+        self.fetched_for_requirements_tentacles_versions = {}
+        self.fetching_requirements = False
+        self.requirements_downloading_event = Event()
+
+    def reset_worker(self):
+        self.errors = []
+
+        self.to_process_tentacle_modules = {}
+        self.processed_tentacles_modules = []
+
+        self.fetched_for_requirements_tentacles = []
+        self.fetched_for_requirements_tentacles_versions = {}
+        self.fetching_requirements = False
+        self.requirements_downloading_event = Event()
+
+    @staticmethod
+    def import_tentacle_config_if_any(target_tentacle_path, replace=False):
+        target_tentacle_config_path = path.join(target_tentacle_path, TENTACLE_CONFIG)
+        for config_file in listdir(target_tentacle_config_path):
+            if config_file.endswith(CONFIG_EXT) and not config_file.endswith(CONFIG_SCHEMA_EXT):
+                target_user_path = path.join(USER_TENTACLE_SPECIFIC_CONFIG_PATH, config_file)
+                if replace or not path.exists(target_user_path):
+                    copyfile(path.join(target_tentacle_config_path, config_file), target_user_path)
+
+    async def refresh_tentacle_config_file(self):
+        available_tentacle_data = await load_tentacle_with_metadata(self.tentacle_path)
+        tentacle_global_config = GlobalTentacleConfiguration()
+        await tentacle_global_config.read_config()
+        await tentacle_global_config.fill_tentacle_config(available_tentacle_data, self.default_tentacle_config)
+        await tentacle_global_config.save_config()
+
+    def log_summary(self):
+        if self.errors:
+            self.logger.info(" *** Error summary: ***")
+            for error in self.errors:
+                self.logger.error(f"Error when handling tentacle: {error}")
+        else:
+            self.logger.info(" *** All tentacles have been successfully processed ***")
+
+    def register_to_process_tentacles_modules(self, to_process_tentacle_data):
+        self.to_process_tentacle_modules = self._get_version_by_tentacle_data(to_process_tentacle_data)
+
+    async def handle_requirements(self, tentacle_data, callback):
+        missing_requirements = self._find_tentacles_missing_requirements(tentacle_data,
+                                                                         self.to_process_tentacle_modules)
+        if missing_requirements:
+            if not self.fetching_requirements:
+                self.fetching_requirements = True
+                await self._fetch_all_available_tentacles()
+            else:
+                await wait_for(self.requirements_downloading_event.wait(), self.TENTACLES_FETCHING_TIMEOUT)
+            await callback(tentacle_data, missing_requirements)
+
+    def confirm_action(self, action):
+        if not self.use_confirm_prompt:
+            return True
+        else:
+            confirmations = ["yes", "ye", "y", "oui", "o", "1"]
```

Comment:
Why not just Y/n ?

Reply:
I changed it to yes / y (was a legacy code from previous version)